elytron-oidc-client/src/main/java/org/wildfly/extension/elytron/oidc/ElytronOidcSubsystemTransformers.java

```diff
@@ -29,25 +35,32 @@ public void registerTransformers(SubsystemTransformerRegistration registration)
 
         ChainedTransformationDescriptionBuilder chainedBuilder = TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(registration.getCurrentSubsystemVersion());
         // 3.0.0 (WildFly 32) to 2.0.0 (WildFly 29)
-//        from3(chainedBuilder);
+        from3(chainedBuilder);
         // 2.0.0 (WildFly 29) to 1.0.0 (WildFly 28)
         from2(chainedBuilder);
 
-        chainedBuilder.buildAndRegister(registration, new ModelVersion[] { VERSION_1_0_0.getVersion() });
+        chainedBuilder.buildAndRegister(registration, new ModelVersion[] { VERSION_2_0_0.getVersion(), VERSION_1_0_0.getVersion() });
     }
 
     private static void from2(ChainedTransformationDescriptionBuilder chainedBuilder) {
         ResourceTransformationDescriptionBuilder builder = chainedBuilder.createBuilder(VERSION_2_0_0.getVersion(), VERSION_1_0_0.getVersion());
         builder.rejectChildResource(PathElement.pathElement(SECURE_SERVER));
     }
-    // TODO: Figure out how to handle the transformer with a preview level attribute being added to the subsystem
-//    private static void from3(ChainedTransformationDescriptionBuilder chainedBuilder) {
-//        ResourceTransformationDescriptionBuilder builder = chainedBuilder.createBuilder(VERSION_3_0_0.getVersion(), VERSION_2_0_0.getVersion());
-//        builder.addChildResource(PathElement.pathElement(SECURE_SERVER))
-//                .getAttributeBuilder();
-//
-//        builder.addChildResource(PathElement.pathElement(SECURE_DEPLOYMENT))
-//                .getAttributeBuilder();
-//    }
+
+    private static void from3(ChainedTransformationDescriptionBuilder chainedBuilder) {
+        ResourceTransformationDescriptionBuilder builder = chainedBuilder.createBuilder(VERSION_3_0_0.getVersion(), VERSION_2_0_0.getVersion());
+
+        builder.addChildResource(PathElement.pathElement(SECURE_SERVER))
+                .getAttributeBuilder()
+                .addRejectCheck(new RejectAttributeChecker.SimpleRejectAttributeChecker(ModelNode.FALSE), SCOPE)
+                .setDiscard(DiscardAttributeChecker.ALWAYS, SCOPE)
```

Comment:
For some reason if I use the following, the scope attribute does not get discarded or rejected properly. It gets sent to version 2, which does not recognize this attribute. 
```
.addRejectCheck( RejectAttributeChecker.DEFINED, SCOPE)
.setDiscard(DiscardAttributeChecker.UNDEFINED, SCOPE)
``` 
So, I had to change it to the reject and discard checkers above. Not entirely sure if this is correct, however.

Reply:
While the `.setDiscard(DiscardAttributeChecker.ALWAYS, SCOPE)` looks okay to me, I am not sure about the reject check. Since from the javadoc of SimpleRejectAttributeChecker: `Rejects the attribute if the value is equal to the specified value.` 

so it seems that this will reject the attribute SCOPE only if the value of it is false. And we want to reject it with any value

Reply:
Thanks @Skyllarr 
I have updated it to 
`.addRejectCheck( RejectAttributeChecker.DEFINED, SCOPE)`